demo.py

```diff
@@ -0,0 +1,418 @@
+from datetime import datetime
+import numpy as np
+import seaborn as sns
+import pandas as pd
+import matplotlib.pyplot as plt
+from utils import (
+    discount_cash_flows
+)
+from financial_calculations.forward_curves import (
+    bootstrap_forward_curve, 
+    calibrate_finer_forward_curve
+)
+from financial_calculations.mbs_cash_flows import (
+    calculate_scheduled_balances,
+    calculate_actual_balances,
+    evaluate_cash_flows
+)
+from financial_models.hull_white import (
+    hull_white_simulate_from_curve
+)
+from financial_models.prepayment import (
+    calculate_pccs,
+    calculate_smms
+)
+
+SETTLE_DATE_IDX = 5
+ORIGINATION_DATE_IDX = 6
+
+def parse_datetime(date_str):
```

Comment:
this could be put into convert_to_datetime with logic based on
if string:
    this logic
else:
     date like logic